File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.cs
public class Solution {
    public string GcdOfStrings(string str1, string str2) {
        if((str1 + str2) != (str2 + str1))
            return "";
                
        int nLen1 = str1.Length;
        int nLen2 = str2.Length;
        
        while(nLen2 != 0)
        {
            int nTemp = nLen2;
            nLen2 = nLen1 % nLen2;
            nLen1 = nTemp;
        }
        
        return str1.Substring(0, nLen1);
    }
}

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        if(str1+str2 != str2 + str1)
            return "";
        
        int nLen1 = str1.length();
        int nLen2 = str2.length();
        
        while(nLen2 != 0)
        {
            int nTemp = nLen2;
            nLen2 = nLen1 % nLen2;
            nLen1 = nTemp;
        }
        
        return str1.substr(0,nLen1);
    }
};